inc/azure_c_util/interlocked_if.h

```diff
@@ -0,0 +1,156 @@
+#ifndef INTERLOCKEDIFIF_H
+#define INTERLOCKEDIFIF_H
+
+#include <stdint.h>
+
+long InterlockedIFAdd( 
```

Comment:
>InterlockedIFAdd [](start = 5, length = 16)

seeing as there is not a struct that gathers all these functions, the implementation is build at link time (likely by selecting one or the other .c files that implement it).

why is it an interface then... at all? It can be as easily a module with different implementations, header stays the same across win32/linux/apple.

asking because other such mechanism that are chosen per OS, are not with interfaces. Example: timer, lock, threadapi. None of them carries "IF" in the exported names. 

I guess it is a matter of poetry, but I don't see the reason why this is an interface, when there can only be 1 implementation / OS and never at the same time 2 such interfaces (different) in the same program.

Reply:
I added "IF" in the names to keep the names as similar to the original Interlocked api while also differentiating them to suggest that these new functions work cross platform. Open to suggestions for other naming schemes.